micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/TimeWindowFixedBoundaryHistogram.java

```diff
@@ -61,7 +61,7 @@ public TimeWindowFixedBoundaryHistogram(Clock clock, DistributionStatisticConfig
             .getHistogramBuckets(supportsAggregablePercentiles);
 
         Boolean percentileHistogram = distributionStatisticConfig.isPercentileHistogram();
-        if (percentileHistogram != null && percentileHistogram) {
+        if (percentileHistogram != null && percentileHistogram && supportsAggregablePercentiles) {
```

Comment:
This is the only change to main code. This makes sure we don't add buckets unless they will be used. If a registry does not `supportsAggregablePercentiles` then it should not do anything when `percentileHistogram` is enabled.